goon/code/datums/browserOutput.dm

```diff
@@ -221,7 +221,7 @@ For the main html chat area
 
 	// Either an atom or somebody fucked up and is gonna get a runtime, which I'm fine with.
 	var/atom/A = obj
-	var/key = "[istype(A.icon, /icon) ? "\ref[A.icon]" : A.icon]:[A.icon_state]"
+	var/key = ("[A.icon]" || "\ref[A.icon]")+":[A.icon_state]"
 	if (!bicon_cache[key]) // Doesn't exist, make it.
 		var/icon/I = icon(A.icon, A.icon_state, SOUTH, 1)
 		if (ishuman(obj)) // Shitty workaround for a BYOND issue.
```

Comment:
Icons read from a file are inserted into a string as the file name. Icons generated at runtime aren't inserted into strings. The BYOND issue is with icons generated at runtime, and that's the only way to determine which they are.